### ISSUE TITLE
Lag Issues

### ISSUE THREAD

**Issue by tignif** [Epix-Incorporated/Adonis#1180]
- What are the steps that lead to the bug occurring? Or, how frequently does the bug occur?
From the data I collected from the server profiler (10 minutes of collecting data with 40-50 people in the game), it takes up 25726.798 m/s (adonis) of the 219113.152 m/s (root)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/7fd04840-be9a-4eec-9acf-bc449455a0e6)
I get around 150-100 players but as soon as it reaches like 40-50 players my game start to lag and adonis is in 2nd place with usage even though I didn't even use any admin commands when I was collecting this data.
I am running on the latest version too

- What would be the normally expected or desired result?
Not to use as much usage especially if not used

- What is the actual result (the bug)?
It’s because it constantly is being invoked by all players in the game and the invitations take multiple seconds. If you look in the consul during the game server, you will see a lot of errors related to it when people leave

- What is the impact of this bug?
The impact is server optimization

- (Any additional context or information eg. in the form of error messages or screenshots.)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/3cad2469-3b3a-46a1-9035-e674f4e5e6d4)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/a05a2941-2943-435c-841c-b735af1423e3)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/9410f81f-2efb-4c86-ae39-47c5b9d82690)

**Comment by tignif** [comment]
https://images-ext-2.discordapp.net/external/UHuuiWCFJabF3r4AZ7j4Sv05RYXb71TI2SPrmfL5G7I/https/i.imgur.com/p0WdgM6.png

every time someone loads, or updates the cache level
it freezes for 0.1-0.2 secs
but if I have 100 players it fucks my game

**Comment by watameln** [comment]
turn off anticheat you dont need it.

**Comment by tignif** [comment]
Anticheat is turned off

**Comment by kaiserandaxl** [comment]
Tried turning off clientcheck?

**Comment by tignif** [comment]
I will post the settings I have disabled when I was collecting data
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/120e893f-d492-4624-b2e4-94e385902b6b)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/f15d5975-59a2-4da9-8830-775ef1511c6f)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/a505fcb6-cb68-4d2a-991c-8c4a93fd6086)
![image](https://github.com/Epix-Incorporated/Adonis/assets/39073631/5461bea3-aebf-478e-954c-d82a84c8013f)

**Comment by kaiserandaxl** [comment]
I might need to see the game myself because Adonis has to work differently depending on what is in the game

**Comment by karl-police** [comment]
Does your screen freeze?

**Comment by tignif** [comment]
Yes when the lag spike occurs, the ping goes up and my screen freezes 

the game is https://www.roblox.com/games/10573562821/V2-San-Junipero-State-Prison

**Comment by tignif** [comment]
add me @tignif discord, i can whitelist you in the game

**Comment by karl-police** [comment]
They said they would share how to fix it, would be helpful so that others can fix it as well.

**Comment by tignif** [comment]
Set max logs from 5000 to 0 that your fix

**Comment by ccuser44** [comment]
Datastores may be the cause of a lot of Adonis lag issues
![image](https://github.com/Epix-Incorporated/Adonis/assets/68124053/3a1ed8de-3e94-468a-b337-a805dca9d3bc)
![image](https://github.com/Epix-Incorporated/Adonis/assets/68124053/d9dd3685-eb0f-4448-b5e5-a8574c7febc3)

**Comment by karl-police** [comment]
> Set max logs from 5000 to 0 that your fix

Hm

**Comment by ccuser44** [comment]
Coasterteam may have fixed this via converting logs to linkedlists

**Comment by karl-police** [comment]
> Coasterteam may have fixed this via converting logs to linkedlists

linkedlists?

**Comment by WalkerOfBacon** [comment]
> > Coasterteam may have fixed this via converting logs to linkedlists
> 
> linkedlists?



> LinkedLists if you didn't know work as a list of nodes which make insertion of nodes a very simple O(1) operation, compared to a table.insert where it may be O(n) depending on table size after the element insertion index. This is an experiment, but it will likely go in!

tl;dr: logging performance improvement

**Comment by ccuser44** [comment]
[![image](https://github.com/Epix-Incorporated/Adonis/assets/68124053/1ef7793a-a00b-466b-9985-938347ee659b)](https://github.com/Epix-Incorporated/Adonis/blob/727aba527298aad2a0abc96f4f3eb367224d3d6d/MainModule/Server/Core/Process.lua#L299C6-L299C6)

This just seems like it's caused by remotes. No idea why though.

Do `:remotelogs` to see why

**Comment by kaiserandaxl** [comment]
This has been fixed in the new version, if lag occurs from it make another bug report please.

@Dimenpsyonal close the issue please, it has been resolved
